configuration/redisnative/redis.go

```diff
@@ -0,0 +1,300 @@
+
+func (r *ConfigurationStore) newFailoverClient(m metadata) *redis.Client {
+	opts := &redis.FailoverOptions{
+		MasterName:      r.metadata.sentinelMasterName,
+		SentinelAddrs:   []string{r.metadata.host},
+		DB:              defaultDB,
+		MaxRetries:      m.maxRetries,
+		MaxRetryBackoff: m.maxRetryBackoff,
+	}
+
+	/* #nosec */
+	if m.enableTLS {
+		opts.TLSConfig = &tls.Config{
+			InsecureSkipVerify: m.enableTLS,
+		}
+	}
+
+	return redis.NewFailoverClient(opts)
+}
+
+func (r *ConfigurationStore) getConnectedSlaves() (int, error) {
+	res, err := r.client.DoContext(context.Background(), "INFO", "replication").Result()
+	if err != nil {
+		return 0, err
+	}
+
+	// Response example: https://redis.io/commands/info#return-value
+	// # Replication\r\nrole:master\r\nconnected_slaves:1\r\n
+	s, _ := strconv.Unquote(fmt.Sprintf("%q", res))
+	if len(s) == 0 {
+		return 0, nil
+	}
+
+	return r.parseConnectedSlaves(s), nil
+}
+
+func (r *ConfigurationStore) parseConnectedSlaves(res string) int {
+	infos := strings.Split(res, infoReplicationDelimiter)
+	for _, info := range infos {
+		if strings.Contains(info, connectedSlavesReplicas) {
+			parsedReplicas, _ := strconv.ParseUint(info[len(connectedSlavesReplicas):], 10, 32)
+
+			return int(parsedReplicas)
+		}
+	}
+
+	return 0
+}
+
+func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
+	keys := req.Keys
+	var err error
+	if len(keys) == 0 {
+		// get all keys
+		if keys, err = r.client.Keys("*").Result(); err != nil {
+			r.logger.Errorf("failed to all keys, error is %s", err)
+		}
+	}
+
+	items := make([]*configuration.Item, 0, 16)
+
+	// query by keys
+	for _, redisKey := range keys {
+		item := &configuration.Item{
+			Metadata: map[string]string{},
+		}
+
+		redisValue, err := r.client.Get(redisKey).Result()
+		if err != nil {
+			return &configuration.GetResponse{}, fmt.Errorf("fail to get configuration for redis key=%s, error is %s", redisKey, err)
+		}
+		val, version := internal.GetRedisValueAndVersion(redisValue)
+		item.Key = redisKey
+		item.Version = version
+		item.Value = val
+
+		if item.Value != "" {
+			// Hash key/value exist and "Content" filed in Hash value exist
+			items = append(items, item)
+		}
+	}
+
+	return &configuration.GetResponse{
+		Items: items,
+	}, nil
+}
+
+func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) error {
+	if len(req.Keys) == 0 {
+		go r.doSubscribe(ctx, req, handler, "__keyspace@0__:*")
+		return nil
+	}
+	for _, k := range req.Keys {
+		go r.doSubscribe(ctx, req, handler, fmt.Sprintf("__keyspace@0__:%s", k))
+	}
+	return nil
+}
+
+func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, redisChannel4revision string) {
+	// enable notify-keyspace-events by redis Set command
+	r.client.ConfigSet("notify-keyspace-events", "KA")
```

Comment:
Is it documented?

Reply:
It's needed to add this configuration for Redis Server,  otherwise, the listening of keyspace cannot be enabled. 
I think I should add this part to docs.